github/github_test.go

```diff
@@ -105,6 +105,16 @@ func testHeader(t *testing.T, r *http.Request, header string, want string) {
 	}
 }
 
+func testCanonicalHeader(t *testing.T, r *http.Request, header string, want string) {
+	canonical := http.CanonicalHeaderKey(header)
+	for _, h := range r.Header[canonical] {
+		if h == want {
+			return
+		}
+	}
+	t.Errorf("Multiple header get, wanted %s but not found for header %s", want, header)
+}
```

Comment:
Gotcha. So it sounds like you do want to test all header values, not just the first one, at least for this particular test.

I wonder how it was that all other tests got away with checking just the first header.

Well, since this is needed, then you should keep this func. While it's possible to change `testHeader` to test for all header values rather than the first, that would make all other tests that use it less strict than they are now, which may not be desired. Unless @willnorris has some comments on this.

Unfortunately, `testHeader` is not documented, so it's not obvious if its intention was to check all header values or only the first. I don't feel comfortable suggesting you change it without knowing that.

But if you keep `testCanonicalHeader`, a suggestion I'll make is 2 things:

1. rename it to something more descriptive, like `testHeaderAny` or `testHeaderMultiple`, and
2. document its intended behavior, so that people writing tests in the future will have an easier time choosing which of `testHeader` or `testHeaderAny` to use. It's not easy to figure out the difference in their behavior from just looking their source code.